code/datums/components/boomerang.dm

```diff
@@ -0,0 +1,89 @@
+///The cooldown period between last_boomerang_throw and it's methods of implementing a rebound proc.
+#define BOOMERANG_REBOUND_INTERVAL (1 SECONDS)
+/**
+ * If an object is given the boomerang component, it should be thrown back to the thrower after either hitting it's target, or landing on the thrown tile.
+ * Thrown objects should be thrown back to the original thrower with this component, a number of tiles defined by boomerang_throw_range.
+ */
+/datum/component/boomerang
+	///How far should the boomerang try to travel to return to the thrower?
+	var/boomerang_throw_range = 3
+	///If this boomerang is thrown, does it re-enable the throwers throw mode?
+	var/thrower_easy_catch_enabled = FALSE
+	///This cooldown prevents our 2 throwing signals from firing too often based on how we implement those signals within thrown impacts.
+	var/last_boomerang_throw = 0
+
+/datum/component/boomerang/Initialize(boomerang_throw_range, thrower_easy_catch_enabled)
+	. = ..()
+	if(!isitem(parent)) //Only items support being thrown around like a boomerang, feel free to make this apply to humans later on.
+		return COMPONENT_INCOMPATIBLE
+
+	//Assignments
+	if(boomerang_throw_range)
+		src.boomerang_throw_range = boomerang_throw_range
+	if(thrower_easy_catch_enabled)
+		src.thrower_easy_catch_enabled = thrower_easy_catch_enabled
```

Comment:
Should be no need to check the values like this unless you're transferring the component
```suggestion
	src.boomerang_throw_range = boomerang_throw_range
	src.thrower_easy_catch_enabled = thrower_easy_catch_enabled
```